app/code/Magento/Customer/Model/Customer.php

```diff
@@ -886,7 +886,7 @@ public function getGroupId()
                 ScopeInterface::SCOPE_STORE,
                 $storeId
             );
-            $this->setData('group_id', $groupId);
+            $this->setData('group_id', (int)$groupId);
         }
         return $this->getData('group_id');
```

Comment:
Why haven't you changed added casting from string to int here?

Reply:
I'm not getting you, will you please elaborate?

Reply:
@Yogeshks he is talking about the return. The return should be type casted.

Reply:
Hi @ferrazzuk
Thanks for the correction, I'll update my commit accordingly.

Reply:
@Yogeshks no worries do that for the other files you also changed. Thank you for contributing

Reply:
@Yogeshks can you move the type casting to the return instead?